### ISSUE TITLE
running main_ce.py does not work

### ISSUE THREAD

**Issue by gr986** [HobbitLong/SupContrast#71]
(pytorch) D:\g\project\333>python SupContrast\main_ce.py
Files already downloaded and verified
Traceback (most recent call last):
  File "SupContrast\main_ce.py", line 347, in <module>
    main()
  File "SupContrast\main_ce.py", line 316, in main
    loss, train_acc = train(train_loader, model, criterion, optimizer, epoch, opt)
  File "SupContrast\main_ce.py", line 223, in train
    output = model(images)
  File "C:\Users\11708\Desktop\minikangda\envs\pytorch\lib\site-packages\torch\nn\modules\module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "D:\g\project\333\SupContrast\resnet_big.py", line 198, in forward
    return self.fc(self.encoder(x))
  File "C:\Users\11708\Desktop\minikangda\envs\pytorch\lib\site-packages\torch\nn\modules\module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "D:\g\project\333\SupContrast\resnet_big.py", line 118, in forward
    out = self.layer1(out)
  File "C:\Users\11708\Desktop\minikangda\envs\pytorch\lib\site-packages\torch\nn\modules\module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "C:\Users\11708\Desktop\minikangda\envs\pytorch\lib\site-packages\torch\nn\modules\container.py", line 100, in forward
    input = module(input)
  File "C:\Users\11708\Desktop\minikangda\envs\pytorch\lib\site-packages\torch\nn\modules\module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "D:\g\project\333\SupContrast\resnet_big.py", line 66, in forward
    out += self.shortcut(x)
  File "C:\Users\11708\Desktop\minikangda\envs\pytorch\lib\site-packages\torch\nn\modules\module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "C:\Users\11708\Desktop\minikangda\envs\pytorch\lib\site-packages\torch\nn\modules\container.py", line 100, in forward
    input = module(input)
  File "C:\Users\11708\Desktop\minikangda\envs\pytorch\lib\site-packages\torch\nn\modules\module.py", line 550, in __call__
    result = self.forward(*input, **kwargs)
  File "C:\Users\11708\Desktop\minikangda\envs\pytorch\lib\site-packages\torch\nn\modules\conv.py", line 353, in forward
    return self._conv_forward(input, self.weight)
  File "C:\Users\11708\Desktop\minikangda\envs\pytorch\lib\site-packages\torch\nn\modules\conv.py", line 350, in _conv_forward
    self.padding, self.dilation, self.groups)
RuntimeError: CUDA out of memory. Tried to allocate 256.00 MiB (GPU 0; 2.00 GiB total capacity; 1.15 GiB already allocated; 21.82 MiB free; 1.26 GiB reserved in total by PyTorch)

**Comment by HobbitLong** [comment]
Basically your GPU memory is not enough, try to use GPU with larger memory, or lower the batch size (together  with reducing the learning rate linearly with batch size)

**Comment by gr986** [comment]
Thanks~




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "HobbitLong/SupContrast"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年3月23日(星期二) 下午4:05
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [HobbitLong/SupContrast] running main_ce.py does not work (#71)





 
Basically your GPU memory is not enough, try to use GPU with larger memory, or lower the batch size (together  with reducing the learning rate linearly with batch size)
